CMakeLists.txt

```diff
@@ -121,12 +122,25 @@ add_compile_definitions(
     QT_USE_QSTRINGBUILDER
 )
 
+if(BUILD_WASM)
+    set(BUILD_TESTING OFF)
+    set(Rust_CARGO_TARGET wasm32-unknown-emscripten)
```

Comment:
If we could figure out the `qt_add_executable` different then maybe we just add the `CMAKE_LINK_LIBRARY_USING_WHOLE_ARCHIVE` here as then there would be no need for any `BUILD_WASM` in the examples?

Depends if we want to allow folks to build the examples standalone for wasm too i guess ? Or whether we just document all these difference in a book page?